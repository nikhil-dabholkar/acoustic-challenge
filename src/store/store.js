import rootReducer from '../reducer/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';

export const store = createStore(rootReducer, applyMiddleware(thunk))