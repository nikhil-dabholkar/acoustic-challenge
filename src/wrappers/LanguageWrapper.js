import React from 'react'
import { IntlProvider } from 'react-intl'
import enUS from '../i18n/en_US.json'
import plPL from '../i18n/pl_PL.json'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
function LanguageWrapper ({ children, language }) {
  let languageMessages = null
  switch (language.selectedLanguage.label) {
    case 'English':
      languageMessages = enUS
      break
    case 'Polish':
      languageMessages = plPL
      break
    default:
      languageMessages = enUS
      break
  }
  return (
    <IntlProvider locale={language.selectedLanguage.locale} messages={languageMessages}>
      <div data-testid="languageWrapper">
        {children}
      </div>
    </IntlProvider>
  )
}

const mapStateToProps = state => {
  return ({
    language: state.languageReducer
  })
}

LanguageWrapper.propTypes = {
  children: PropTypes.node.isRequired,
  language: PropTypes.object
}

export default connect(mapStateToProps)(LanguageWrapper)
