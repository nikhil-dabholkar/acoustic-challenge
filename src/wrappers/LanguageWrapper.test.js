import React from 'react'
import LanguageWrapper from './LanguageWrapper'
import { store } from '../store/store'
import { Provider } from 'react-redux'
import { render, cleanup, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import enUS from '../i18n/en_US.json'

afterEach(cleanup)

function renderWithRouter (
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  }
}

function rendererWithRouter (
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  return {
    ...renderer.create(<Router history={history}>{ui}</Router>),
    history
  }
}

describe('LanguageWrapper', () => {
  test('Check if LanguageWrapper Renders on /home', async () => {
    const route = '/home'
    renderWithRouter(<Provider store={store}><LanguageWrapper locale="en" messages={enUS}></LanguageWrapper></Provider>, { route })
    const lazyElement = await screen.findByTestId('languageWrapper')
    expect(lazyElement).toBeInTheDocument()
  })

  test('Check if LanguageWrapper Renders on /articles', async () => {
    const route = '/articles'
    renderWithRouter(<Provider store={store}><LanguageWrapper locale="en" messages={enUS}></LanguageWrapper></Provider>, { route })
    const lazyElement = await screen.findByTestId('languageWrapper')
    expect(lazyElement).toBeInTheDocument()
  })

  test('Check if LanguageWrapper Renders on /articles/fa9519d5-0363-4b8d-8e1f-627d802c08a8', async () => {
    const route = '/articles/fa9519d5-0363-4b8d-8e1f-627d802c08a8'
    renderWithRouter(<Provider store={store}><LanguageWrapper locale="en" messages={enUS}></LanguageWrapper></Provider>, { route })
    const lazyElement = await screen.findByTestId('languageWrapper')
    expect(lazyElement).toBeInTheDocument()
  })

  test('Check if LanguageWrapper Renders on /articles/invalid_item', async () => {
    const route = '/articles/invalid_item'
    renderWithRouter(<Provider store={store}><LanguageWrapper locale="en" messages={enUS}></LanguageWrapper></Provider>, { route })
    const lazyElement = await screen.findByTestId('languageWrapper')
    expect(lazyElement).toBeInTheDocument()
  })

  test('Check if LanguageWrapper Renders on /', async () => {
    const route = '/'
    renderWithRouter(<Provider store={store}><LanguageWrapper locale="en" messages={enUS}></LanguageWrapper></Provider>, { route })
    const lazyElement = await screen.findByTestId('languageWrapper')
    expect(lazyElement).toBeInTheDocument()
  })

  test('Check if matches LanguageWrapper Snapshot', async () => {
    const route = '/home'
    const tree = rendererWithRouter(<Provider store={store}><LanguageWrapper locale="en" messages={enUS}></LanguageWrapper></Provider>, { route }).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
