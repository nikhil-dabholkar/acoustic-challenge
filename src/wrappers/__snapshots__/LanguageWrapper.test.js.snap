// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageWrapper Check if matches LanguageWrapper Snapshot 1`] = `
<div
  data-testid="languageWrapper"
/>
`;
